homeassistant/components/canary/__init__.py

```diff
@@ -137,18 +135,15 @@ async def _async_update_listener(hass: HomeAssistantType, entry: ConfigEntry) ->
 
 
 class CanaryData:
-    """Get the latest data and update the states."""
+    """Manages the data retrieved from Canary API."""
 
     def __init__(self, username, password, timeout):
         """Init the Canary data object."""
-
         self._api = Api(username, password, timeout)
```

Comment:
This also has side effects. Preferably instantiate the api first and pass that instance to the CanaryData as a parameter.

Reply:
valid point, that kinda sets it up better for DataUpdateCoordinator even.